graphene_federation/service.py

```diff
@@ -90,14 +101,57 @@ def get_sdl(schema: Schema) -> str:
 
     # Get various objects that need to be amended
     extended_types = get_extended_types(schema)
+    shareable_types = get_shareable_types(schema)
+    inaccessible_types = get_inaccessible_types(schema)
     provides_parent_types = get_provides_parent_types(schema)
+    provides_fields = get_provides_fields(schema)
     entities = get_entities(schema)
-
-    # Add fields directives (@external, @provides, @requires)
-    for entity in set(provides_parent_types.values()) | set(extended_types.values()):
+    shareable_fields = get_shareable_fields(schema)
+    tagged_fields = get_tagged_fields(schema)
+    inaccessible_fields = get_inaccessible_fields(schema)
+    required_fields = get_required_fields(schema)
+    external_fields = get_external_fields(schema)
+    override_fields = get_override_fields(schema)
+
+    _schema_import = []
+
+    if extended_types:
+        _schema_import.append('"@extends"')
+    if external_fields:
+        _schema_import.append('"@external"')
+    if entities:
+        _schema_import.append('"@key"')
+    if inaccessible_types or inaccessible_fields:
+        _schema_import.append('"@inaccessible"')
+    if override_fields:
+        _schema_import.append('"@override"')
+    if provides_parent_types or provides_fields:
+        _schema_import.append('"@provides"')
+    if required_fields:
+        _schema_import.append('"@requires"')
+    if shareable_types or shareable_fields:
+        _schema_import.append('"@shareable"')
+    if tagged_fields:
+        _schema_import.append('"@tag"')
+    schema_import = ", ".join(_schema_import)
+    _schema = f'extend schema @link(url: "https://specs.apollo.dev/federation/v2.0", import: [{schema_import}])\n'
```

Comment:
To clarify: would this PR remove support for `Federation<2` due to the addition of this? 

Given the rest of the implementation, one should be fine on `1.x` by simply restricting fields to use the old directives, right?
 I want to make sure we don't push support for graphene `v3` and then immediately force everyone to upgrade to gateway `v2`, if there's a simple way to avoid that.

Reply:
in strawberry we made this opt-in, I think we should do the same here